This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Sun, 24 Nov 2019 13:00:00 +0100
Subject: [PATCH 1/1] Allow extra information within the target triple

For example: x86_64-w64-mingw32.static

Upstream-Status: Pending [https://github.com/libvips/build-win64-mxe/issues/37]

diff --git a/wrappers/native-wrapper.h b/wrappers/native-wrapper.h
index 1111111..2222222 100644
--- a/wrappers/native-wrapper.h
+++ b/wrappers/native-wrapper.h
@@ -37,6 +37,7 @@ typedef char TCHAR;
 #define _T(x) x
 #define _tcsrchr strrchr
 #define _tcschr strchr
+#define _tcsstr strstr
 #define _tcsdup strdup
 #define _tcscpy strcpy
 #define _tcslen strlen
@@ -198,9 +199,9 @@ static inline void split_argv(const TCHAR *argv0, const TCHAR **dir_ptr, const T
     }
 #endif
     basename = _tcsdup(basename);
-    TCHAR *period = _tcschr(basename, '.');
-    if (period)
-        *period = '\0';
+    TCHAR *extension = _tcsstr(basename, ".exe");
+    if (extension)
+        *extension = '\0';
     TCHAR *target = _tcsdup(basename);
     TCHAR *dash = _tcsrchr(target, '-');
     const TCHAR *exe = basename;
